---
layout: paper
title: "High-resolution mapping of copy-number alterations with massively parallel sequencing"
year: "2009"
shortref: "Chiang et al. Nature Methods 2009"
nickname: SegSeq
journal: "Nature Methods"
volume: 6
issue: 1
pages: 99-103
authors: "Chiang DY, Getz G, Jaffe DB, O'Kelly MJ, Zhao X, Carter SL, Russ C, Nusbaum C, Meyerson M, Lander ES"
first_authors: "Chiang DY, Getz G"
senior_authors: "Lander ES"
corresponding_authors: "Lander ES"
image: /assets/images/papers/Chiang2009_Fig1.png
pdf:
pdflink:
github:
pmid: 19043412
pmcid: PMC2630795
doi: 10.1038/nmeth.1276
category: paper
published: true
peerreview: true
tags: [Copy Number Calling]
---

# Abstract

Cancer results from somatic alterations in key genes, including point mutations, copy number alterations and structural rearrangements. A powerful way to discover cancer-causing genes is to identify genomic regions that show recurrent copy-number alterations (gains and losses) in tumor genomes. Recent advances in sequencing technologies suggest that massively parallel sequencing may provide a feasible alternative to DNA microarrays for detecting copy-number alterations. Here, we present: (i) a statistical analysis of the power to detect copy-number alterations of a given size; (ii) SegSeq, an algorithm to identify chromosomal breakpoints using massively parallel sequence data; and (iii) analysis of experimental data from three matched pairs of tumor and normal cell lines. We show that a collection of ∼14 million aligned sequence reads from human cell lines has comparable power to detect events as the current generation of DNA microarrays and has over two-fold better precision for localizing breakpoints (typically, to within ∼1 kb).

