---
layout: paper
title: "Sensitive detection of somatic point mutations in impure and heterogeneous cancer samples"
year: "2013"
shortref: "Cibulskis et al. Nature Biotechnology 2013"
nickname: SegSeq
journal: "Nature Biotechnology"
volume:
issue: 3
pages: 213-219
authors: "Cibulskis K, Lawrence MS, Carter SL, Sivachenko A, Jaffe D, Sougnez C, Gabriel S, Meyerson M, Lander ES, Getz G"
image: /assets/images/papers/Cibulskis2013_Fig1.png
pdf:
pdflink: https://www.nature.com/articles/nbt.2514
github:
pmid: 23396013
pmcid: PMC3833702
doi: 10.1038/nbt.2514
category: paper
published: true
peerreview: true
tags: [Mutation Calling]
---
{% include JB/setup %}

# Abstract

Detection of somatic point substitutions is a key step in characterizing the cancer genome. Mutations in cancer are rare (0.1â€“100/Mb) and often occur only in a subset of the sequenced cells, either due to contamination by normal cells or due to tumor heterogeneity. Consequently, mutation calling methods need to be both specific, avoiding false positives, and sensitive to detect clonal and sub-clonal mutations. The decreased sensitivity of existing methods for low allelic fraction mutations highlights the pressing need for improved and systematically evaluated mutation detection methods. Here we present MuTect, a method based on a Bayesian classifier designed to detect somatic mutations with very low allele-fractions, requiring only a few supporting reads, followed by a set of carefully tuned filters that ensure high specificity. We also describe novel benchmarking approaches, which use real sequencing data to evaluate the sensitivity and specificity as a function of sequencing depth, base quality and allelic fraction. Compared with other methods, MuTect has higher sensitivity with similar specificity, especially for mutations with allelic fractions as low as 0.1 and below, making MuTect particularly useful for studying cancer subclones and their evolution in standard exome and genome sequencing data.


