---
layout: paper
title: "PRADA: pipeline for RNA sequencing data analysis"
year: "2014"
shortref: "Torres-Carcia et al. Bioinformatics 2014"
nickname: prada
journal: "Bioinformatics"
volume: 30
issue: 15
pages: 2224-6
authors: "Torres-García W, Zheng S, Sivachenko A, Vegesna R, Wang Q, Yao R, Berger MF, Weinstein JN, Getz G, Verhaak RG"
first_authors: "Torres-García W, Zheng S, Sivachenko A"
senior_authors: "Getz G, Verhaak RG"
corresponding_authors: "Torres-García W, Zheng S, Sivachenko A"
image: /assets/images/papers/default-paper.png
pdflink: 
fulltext:
doi: 10.1093/bioinformatics/btu169
pmid: 24695405
pmcid: PMC4103589
category: paper
published: true
peerreview: true
tags: [RNAseq, TCGA, glioblastoma, renal clear cell]
---

# Abstract

Technological advances in high-throughput sequencing necessitate improved computational tools for processing and analyzing large-scale datasets in a systematic automated manner. For that purpose, we have developed PRADA (Pipeline for RNA-Sequencing Data Analysis), a flexible, modular and highly scalable software platform that provides many different types of information available by multifaceted analysis starting from raw paired-end RNA-seq data: gene expression levels, quality metrics, detection of unsupervised and supervised fusion transcripts, detection of intragenic fusion variants, homology scores and fusion frame classification. PRADA uses a dual-mapping strategy that increases sensitivity and refines the analytical endpoints. PRADA has been used extensively and successfully in the glioblastoma and renal clear cell projects of The Cancer Genome Atlas program.



