---
layout: paper
title: "Pan-cancer analysis of whole genomes"
year: "2020"
shortref: "PCAWG Consortium Nature 2020"
nickname: pcawg-flagship
journal: "Nature"
volume: 578
issue: 7793
pages: 82-93
authors: "Getz G, Rheinbay E, PCAWG Consortium"
image: /assets/images/papers/PCAWG2020_Fig1.png
pdf:
pdflink: https://www.nature.com/articles/s41586-020-1969-6.pdf
pmid: 32025007
pmcid: PMC7025898
doi: 10.1038/s41586-020-1969-6
category: paper
published: true
peerreview: true
tags: [pcawg]
---
{% include JB/setup %}

# Abstract

Cancer is driven by genetic change, and the advent of massively parallel sequencing has enabled systematic documentation of this variation at the whole-genome scale. Here we report the integrative analysis of 2,658 whole-cancer genomes and their matching normal tissues across 38 tumour types from the Pan-Cancer Analysis of Whole Genomes (PCAWG) Consortium of the International Cancer Genome Consortium (ICGC) and The Cancer Genome Atlas (TCGA). We describe the generation of the PCAWG resource, facilitated by international data sharing using compute clouds. On average, cancer genomes contained 4â€“5 driver mutations when combining coding and non-coding genomic elements; however, in around 5% of cases no drivers were identified, suggesting that cancer driver discovery is not yet complete. Chromothripsis, in which many clustered structural variants arise in a single catastrophic event, is frequently an early event in tumour evolution; in acral melanoma, for example, these events precede most somatic point mutations and affect several cancer-associated genes simultaneously. Cancers with abnormal telomere maintenance often originate from tissues with low replicative activity and show several mechanisms of preventing telomere attrition to critical levels. Common and rare germline variants affect patterns of somatic mutation, including point mutations, structural variants and somatic retrotransposition. A collection of papers from the PCAWG Consortium describes non-coding mutations that drive cancer beyond those in the TERT promoter; identifies new signatures of mutational processes that cause base substitutions, small insertions and deletions and structural variation; analyses timings and patterns of tumour evolution; describes the diverse transcriptional consequences of somatic mutation on splicing, expression levels, fusion genes and promoter activity; and evaluates a range of more-specialized features of cancer genomes

